### ISSUE TITLE
Arbitrary order for EntityListener

### ISSUE THREAD

**Issue by Nauktis** [libgdx/ashley#154]
The order in which the EntityListeners are executed is arbitrary (entries of an ObjectMap).
Let's take an example.
I have two components, A and B.
I also have two EntityListeners. One that tracks changes in Familly.all(A.class) and the other that tracks changes in Familly.all(B.class).
When removing an entity that contains a Component A and a Component B from the engine, both listeners are called. The order in which they are called is arbitrary (it is basically the order in which the Engine got to know about the Families).

I have many cases where it is really important that one entity listener is executed before the other. 
Long story short, there should be a notion of priority for entity listeners.
